fullstack-core/fullstack-infrastructure-api/src/main/java/com/hedera/fullstack/infrastructure/api/model/WorkloadReplica.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.fullstack.infrastructure.api.model;
+
+import java.util.List;
+
+/**
+ * A unique indexed instance of {@link Workload}
+ * @param <T> type of the workload
+ */
+public class WorkloadReplica<T extends Workload> {
+
+    private List<Component> components;
+
+    // global index across all clusters in the NetworkDeployment
+    private final int index;
+
+    public int index() {
+        return index;
+    }
+
+    public WorkloadReplica(List<Component> components, int index) {
```

Comment:
Method arguments for concrete classes should be `final`. Needs javadoc. Needs docs for exceptions thrown and needs safety checks for `null`. 

```suggestion
    public WorkloadReplica(final List<Component> components, final int index) {
```